zingolib/src/blaze/trial_decryptions.rs

```diff
@@ -167,26 +169,59 @@ impl TrialDecryptions {
             let mut orchard_notes_to_mark_position_in_block = Vec::new();
 
             for (transaction_num, compact_transaction) in compact_block.vtx.iter().enumerate() {
+                let mut sapling_notes_to_mark_position_in_tx: Vec<_> = compact_transaction
+                    .outputs
+                    .iter()
+                    .enumerate()
+                    .map(|(i, output)| {
+                        (
+                            i as u32,
+                            TxId::from_bytes(
+                                <[u8; 32]>::try_from(&compact_transaction.hash[..]).unwrap(),
+                            ),
+                            sapling::Node::from_commitment(output.cmstar()).unwrap(),
+                            Retention::Ephemeral,
+                        )
+                    })
+                    .collect();
+                let mut orchard_notes_to_mark_position_in_tx: Vec<_> = compact_transaction
+                    .actions
+                    .iter()
+                    .enumerate()
+                    .map(|(i, output)| {
+                        (
+                            i as u32,
+                            TxId::from_bytes(
+                                <[u8; 32]>::try_from(&compact_transaction.hash[..]).unwrap(),
+                            ),
+                            MerkleHashOrchard::from_commitment(output.cmstar()).unwrap(),
+                            Retention::Ephemeral,
+                        )
+                    })
+                    .collect();
+
                 if let Some(filter) = transaction_size_filter {
                     if compact_transaction.outputs.len() + compact_transaction.actions.len()
                         > filter as usize
                     {
-                        break;
+                        sapling_notes_to_mark_position_in_block
```

Comment:
we use .extend_from_slice here... and again on line 237??

Reply:
We call `continue` on the next line, which skips to the next iteration of the loop

Reply:
aha.

Reply:
so im reading if the transaction is bigger than the filter...

Reply:
then we SKIP trial decryption, BUT we STILL mark_position of every note in it as ephemeral?